src/deprecated.jl

```diff
@@ -48,6 +48,12 @@ extern(x::NotImplemented) = (Base.depwarn(EXTERN_DEPRECATION, :extern); throw(No
 
 @inline extern(x::AbstractThunk) = (Base.depwarn(EXTERN_DEPRECATION, :extern); return extern(unthunk(x)))
 
+function (x::Thunk)()
+    Base.depwarn("`(x::Thunk)()`` is deprecated, use `unthunk(x)`", :call_Thunk)
+    unthunk(x)
+end
 
-
-
+function (x::InplaceableThunk)()
+    Base.depwarn("`(x::InplaceableThunk)()`` is deprecated, use `unthunk(x)`", :call_InplaceableThunk)
+    unthunk(x)
+end
```

Comment:
Could we just replace both of these with
```julia
function (x::AbstractThunk)()
    Base.depwarn("`(x::AbstractThunk)()`` is deprecated, use `unthunk(x)`", :call_AbstractThunk)
    return unthunk(x)
end
```
?

Also, BlueStyle.jl uses explicit `return`

Reply:
You mean, use `AbstractThunk` in the depwarn message for both of them?

Reply:
Good point, should have been something like
```julia
function (x::T)() where T <: AbstractThunk
    Base.depwarn("`(x::T)()`` is deprecated, use `unthunk(x)`", :call_AbstractThunk)
    return unthunk(x)
end
```

Reply:
> Good point, should have been something like [...]

Oh, yes, that's nicer.

Reply:
I don't think we can do that - ChainRulesCore is backward compatible with Julia v1.0, and `(x::AbstractThunk)() = ...` would require >= v1.3, right? And `(x::T)() where T <: AbstractThunk = ...` even v1.6 won't allow.

Reply:
Thanks for checking 1.0. With the risk of belabouring the idea:
```julia
for T in (:Thunk, :InplaceableThunk)
    @eval function (x::$T)()
        Base.depwarn("`(x::" * string($T) * ")()` is deprecated, use `unthunk(x)`", Symbol(:call_, $(T))) 
        return unthunk(x)
    end    
end
```
should work. Feel free to ignore if you think that the current way is better (but please do add `return`)

Reply:
Sure - just updated the PR, using this variant.